src/interpreter.ts

```diff
@@ -25,3 +499,840 @@ export function parseAndEvalExpression(sourceCode: string): EvalResult {
+                // without a value. This is an error only if the return type of the
+                // function is not void
+                if (returns.kind !== "void") {
+                    throwErrorConstEval(
+                        `function ${idText(functionCode.name)} must return a value`,
+                        functionCode.loc,
+                    );
+                } else {
+                    // The function does not return a value.
+                    // We rely on the typechecker so that the function is called as a statement.
+                    // Hence, we can return a dummy null, since the null will be discarded anyway.
+                    return null;
+                }
+            },
+            { names: paramNames, values: argValues },
+        );
+    }
+
+    public interpretStatement(ast: AstStatement): void {
+        switch (ast.kind) {
+            case "statement_let":
+                this.interpretLetStatement(ast);
+                break;
+            case "statement_assign":
+                this.interpretAssignStatement(ast);
+                break;
+            case "statement_augmentedassign":
+                this.interpretAugmentedAssignStatement(ast);
+                break;
+            case "statement_condition":
+                this.interpretConditionStatement(ast);
+                break;
+            case "statement_expression":
+                this.interpretExpressionStatement(ast);
+                break;
+            case "statement_foreach":
+                this.interpretForEachStatement(ast);
+                break;
+            case "statement_repeat":
+                this.interpretRepeatStatement(ast);
+                break;
+            case "statement_return":
+                this.interpretReturnStatement(ast);
+                break;
+            case "statement_try":
+                this.interpretTryStatement(ast);
+                break;
+            case "statement_try_catch":
+                this.interpretTryCatchStatement(ast);
+                break;
+            case "statement_until":
+                this.interpretUntilStatement(ast);
+                break;
+            case "statement_while":
+                this.interpretWhileStatement(ast);
+                break;
+        }
+    }
+
+    public interpretLetStatement(ast: AstStatementLet) {
+        const val = this.interpretExpression(ast.expression);
+        this.envStack.setNewBinding(idText(ast.name), val);
+    }
+
+    public interpretAssignStatement(ast: AstStatementAssign) {
+        const path = ast.path;
+        const exp = ast.expression;
+
+        if (path.kind === "id") {
+            const val = this.interpretExpression(exp);
+            this.envStack.updateBinding(idText(path), val);
+        } else {
+            throwNonFatalErrorConstEval(
+                `only identifiers are currently supported as path expressions`,
```

Comment:
I'm wondering why ESLint does not complain here: you are using backticks for a string that does not have any interpreted TS expressions inside